### PR TITLE
Introduce `VideoFrameReference` archetype and base video visualization on it

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Large chunk of  #7368
---

* Introduce `VideoFrameReference` archetype
   * comes with a bunch of subtypes
* Make the video visualizer use this
   * in the process I did some general iteration on how this visualizer ticks
      * note that I departed from the usual quite encrusted patterns of how to implement visualizers that naturally found its way into here as well. Imho the patterns don't hold up all that well in general (we had to grind over them many times in quick succession) and less so in particular for "special" archetypes like video or mesh (etc.)
* video data loader emits `VideoFrameReference` instead of the adhoc `VideoTick

Still missing for completion of the video frame reference task:
* examples to use this from SDKs (by directly logging video frames)
   * this surely will show that we need more extensions 
* pretty/edit ui for video timestamps
   * @Wumpf are you saying that you can then scrub videos from a blueprint slider?! Yes. That's exactly what will end up happening ;D
* pretty/edit ui for entity paths 
* exposed utilities to SDKs for generation of video frame references

@ reviewers: Special attention please on the fbs definition, thank you!

image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7396?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7396?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/7396)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -42,16 +42,14 @@
     },
     "rust-analyzer.diagnostics.enable": false, // False positives
     "rust-analyzer.showUnlinkedFileNotification": false,
-
     // Uncomment the following options and restart rust-analyzer to get it to check code behind `cfg(target_arch=wasm32)`.
     // Don't forget to put it in a comment again before committing.
```
Comment (emilk) [inline .vscode/settings.json:46]: Did you forget?

Comment (Wumpf) [reply .vscode/settings.json]: there's more of that

File: pixi.toml [hunk 2]
```diff
@@ -502,8 +502,8 @@ scripts = ["pixi/maturin_config.bat"]
 # This is very similar to `pixi run py-build`, and dispatches to maturin by way of PEP621.
 # However, pixi doesn't know how to track the rust dependencies of the python package, so
 # you still need to `pixi run py-build` in the correct environment if you change the rust code.
-rerun-sdk = { path = "rerun_py", editable = true }
-rerun-notebook = { path = "rerun_notebook", editable = true }
+# rerun-sdk = { path = "rerun_py", editable = true }
+# rerun-notebook = { path = "rerun_notebook", editable = true }
```
Comment (emilk) [inline pixi.toml]: It looks like you didn't self-review before opening this PR up for review :T

File: crates/store/re_types/definitions/rerun/datatypes/video_timestamp.fbs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    Invalid = 0,
+
+    /// Presentation timestamp in nanoseconds since the beginning of the video.
+    Nanoseconds = 1 (default),
+
+    // Future values: FrameNr
+}
+
+/// Timestamp inside a [archetypes.AssetVideo].
+struct VideoTimestamp (
```
Comment (jleibs) [inline crates/store/re_types/definitions/rerun/datatypes/video_timestamp.fbs:15]: I would still love to eventually have `time_mode = FrameNumber`.

So calling this a `Timestamp` will create a bit of a strange naming pattern.

Comment (Wumpf) [reply crates/store/re_types/definitions/rerun/datatypes/video_timestamp.fbs:15]: coincidentally I brought this very thing up on Slack expressing a similar (but not very deep) concern. @emilk pointed out that frame numbers can be seen as a form timestamp unit in this context.
very open to other suggestions not sure on this one 🤷  


https://rerunio.slack.com/archives/C041NHU952S/p1725981749912279?thread_ts=1725981199.569259&cid=C041NHU952S

Comment (jprochazk) [reply crates/store/re_types/definitions/rerun/datatypes/video_timestamp.fbs:15]: Some ideas from ChatGPT

```
VideoPosition: Emphasizes that the struct represents a position in the video, which could be either in frames or seconds.
VideoMarker: Suggests a point in the video, adaptable to frames or time.
TimeOrFrame: Makes it clear that the struct can hold either a time value or a frame number.
FrameOrTimeStamp: Explicitly names the two modes.
VideoCoordinate: Represents a general position in time (seconds or frames) as coordinates often imply different dimensions.
MediaTimeUnit: Indicates the struct holds a unit of time in different formats, either frames or seconds.
PlaybackPosition: Focuses on the position in playback, where both time and frame counts are relevant.
TimelineIndex: Suggests an index along a timeline that can be expressed in either frames or time.
```

**Review by jprochazk — APPROVED** [review]
Works great!

Naming is a non-issue imo, we can change it in a follow-up PR if we think `VideoTimestamp` really doesn't work

**Pull request merged**